src/realm/sync/instruction_replication.cpp

```diff
@@ -447,6 +447,13 @@ void SyncReplication::set(const Table* table, ColKey col, ObjKey key, Mixed valu
 {
     TrivialReplication::set(table, col, key, value, variant);
 
+    if (col == table->get_primary_key_column()) {
```

Comment:
@simonask Shouldn't this be fixed in v10?

Reply:
There are several more fixes that should be backported.

Reply:
I would suggest we do thing the other way. Fix bugs in our current releases, and forward port to dev branches.
In any case either create PR's or issues right away.

Reply:
None of these fixes are important enough to justify spend that much time on it.

Reply:
Ok - so you are saying they won't affect end users (enough) and we don't want them released in v10?

Reply:
End users should not be affected at all, so yes. :-)